File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {

        int mx_so_far = nums[0]; //Set the mx to first element
        
        for(int i = 1;i<nums.length;i++){
            if(mx_so_far < i){  //if max<i then we cannot reach to end
                return false;
            }
            
            mx_so_far = Math.max(mx_so_far, nums[i]+i);
            
        }
        
         return true;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_reach=nums[0] #Initialize maximum reachable element to first index
        
        for i in range(1, len(nums)): # Fhir index 1 se loop kro end tak
            if max_reach<i: # agar max_reach less that hai index se then false or can't reach
                return False
            else:
                max_reach=max(max_reach, nums[i]+i) #max_reach me max daal do curr max ka aur index aur element ko add krke jo aaye wo
        return True